sockeye/inference.py

```diff
@@ -1052,32 +1052,38 @@ def __init__(self,
 
         self._update_scores = UpdateScores()
         self._update_scores.initialize(ctx=self.context)
-        self._update_scores.hybridize()
+        self._update_scores.hybridize(static_alloc=True, static_shape=True)
 
-        # topk function used in beam search
         if self.skip_topk:
-            self._top = partial(utils.top1,
-                                offset=self.offset)
+            self._top = partial(utils.top1, offset=self.offset)  # type: Callable
         else:
-            self._top = partial(utils.topk,
-                                k=self.beam_size,
-                                batch_size=self.batch_size,
-                                offset=self.offset,
-                                use_mxnet_topk=self.context != mx.cpu())  # MXNet implementation is faster on GPUs
+            if self.restrict_lexicon:
```

Comment:
We could comment here that we don't use the HybridBlock TopK since vocabulary selection cannot return a static shape for the scores.  We could have a TODO to keep thinking about better solutions for supporting all features in hybridizable blocks for better performance.  The lexicon lookup methods already have a TODO to revisit the implementation when MXNet has better support for set-like operations.

Reply:
Done!